### PR TITLE
xattr

### PR DESCRIPTION
**Pull request by ArsenArsen**

### CODE REVIEW

File: abis/linux/xattr.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * defer to them if present, and provide a best effort (but near-impossible to
+ * change) alternative otherwise.
+ */
+
+/* __USE_KERNEL_XATTR_DEFS is exported when XATTR_* are emitted, and
+ * __UAPI_DEF_XATTR is used to determine the behaviour of that header, if it's
+ * set to 1, the header exports these defines and __USE_KERNEL_XATTR_DEFS.
+ * This applies for pretty much all other defines in libc-compat.h AFAICT.
+ */
+#if !defined(__USE_KERNEL_XATTR_DEFS) && __has_include(<linux/xattr.h>)
```
Comment (Dennisbonke) [inline abis/linux/xattr.h:15]: I don't see where __USE_KERNEL_XATTR_DEFS is defined?

File: sysdeps/managarm/include/abi-bits/xattr.h [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../abis/mlibc/xattr.h
```
Comment (Dennisbonke) [inline sysdeps/managarm/include/abi-bits/xattr.h:1]: Managarm wants to use the Linux ABI, so we should use the Linux ABI here.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]

File: abis/linux/xattr.h (hunk 1 above)
Comment (ArsenArsen) [reply abis/linux/xattr.h:15]: <linux/xattr.h>

File: sysdeps/managarm/include/abi-bits/xattr.h (hunk 2 above)
Comment (ArsenArsen) [reply sysdeps/managarm/include/abi-bits/xattr.h:1]: Save for support for interacting with linux-headers, it's identical

**Review by Geertiebear — CHANGES_REQUESTED** [review]
The branch should be rebased, there are some merge conflicts unfortunately.

File: abis/mlibc/xattr.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#ifndef MLIBC_ABIS_MLIBC_XATTR_H
```
[the remark this answers is not in the document]
Comment (ArsenArsen) [reply abis/mlibc/xattr.h:1]: it's the same as there, iirc this was written before the custom api was deprecated

**Review by qookei — APPROVED** [review]

**Pull request merged**